plugins/download_statistics/plugin_tests/download_statistics_test.py

```diff
@@ -0,0 +1,119 @@
+        for folder in folders:
+            if folder['public'] is True:
+                self.publicFolder = folder
+
+        self.filesDir = os.path.join(ROOT_DIR, 'plugins', 'download_statistics',
+                                     'plugin_tests', 'files')
+
+    def _downloadItem(self, itemId):
+        # Download item through REST api
+        path = '/item/%s/download' % str(itemId)
+        resp = self.request(path, user=self.admin, isJson=False)
+        self.assertStatusOk(resp)
+        self.getBody(resp)
+
+    def _downloadFile(self, fileId):
+        # Download file through REST api
+        path = '/file/%s/download' % str(fileId)
+        resp = self.request(path, user=self.admin, isJson=False)
+        self.assertStatusOk(resp)
+        self.getBody(resp)
+
+    def _checkDownloadsStarted(self, fileId, count):
+        # Test downloadsStarted is equal to count
+        file = self.model('file').load(fileId, force=True)
+        self.assertEqual(file['downloadStatistics']['started'], count,
+                         'Started Downloads count inaccurate')
+
+    def _checkDownloadsRequested(self, fileId, count):
+        # Test downloadsStarted is equal to count
+        file = self.model('file').load(fileId, force=True)
+        self.assertEqual(file['downloadStatistics']['requested'], count,
+                         'Requested Downloads count inaccurate')
+
+    def _checkDownloadsCompleted(self, fileId, count):
+        # Test downloadsStarted is equal to count
+        file = self.model('file').load(fileId, force=True)
+        self.assertEqual(file['downloadStatistics']['completed'], count,
+                         'Completed downloads count inaccurate')
+
+    def testItemAndFileDownload(self):
+        # Create item
+        item = self.model('item').createItem('item1', self.admin, self.publicFolder)
+
+        # Path to test files
+        file1Path = os.path.join(self.filesDir, 'txt1.txt')
+        file2Path = os.path.join(self.filesDir, 'txt2.txt')
+
+        # Upload files to item
+        with open(file1Path, 'rb') as fp:
+            self.model('upload').uploadFromFile(fp, os.path.getsize(file1Path),
+                                                'txt1.txt', parentType='item',
+                                                parent=item, user=self.admin)
+            self.model('upload').uploadFromFile(fp, os.path.getsize(file2Path),
+                                                'txt2.txt', parentType='item',
+                                                parent=item, user=self.admin)
+
+        # Download item, and its files, several times and ensure downloads are recorded
+        files = list(self.model('item').childFiles(item=item))
+        self.assertTrue(files)
+        for n in range(0, 5):
+            self._downloadItem(item['_id'])
+            for file in files:
+                self._downloadFile(file['_id'])
+        for file in files:
+            self._checkDownloadsStarted(file['_id'], 10)
```

Comment:
These assertions should be tested by retrieving the properties via the API (rather than a database query), to ensure they're being exposed properly.